### PR TITLE
minor optimizations

### PR DESCRIPTION
**Pull request by 0xBeans**
## Motivation

An honest days work.

Small optimization of unchecked , changing to ++i or --i rather than += 1 or i -=1

### CODE REVIEW

File: reference/lib/ReferenceOrderCombiner.sol [hunk 1]
```diff
@@ -481,8 +481,10 @@ contract ReferenceOrderCombiner is
 
             // If offerer and recipient on the execution are the same...
             if (execution.item.recipient == execution.offerer) {
-                // increment total filtered executions.
-                totalFilteredExecutions += 1;
+                unchecked {
```
Comment (0age) [inline reference/lib/ReferenceOrderCombiner.sol:484]: Can you leave a comment as to why unchecked is safe here above the block and others? (i.e. that it is infeasible to increment that high)

**Review by 0age — CHANGES_REQUESTED** [review]

File: reference/lib/ReferenceOrderCombiner.sol (hunk 1 above)
Comment (0xBeans) [reply reference/lib/ReferenceOrderCombiner.sol:484]: @0age Done.

File: reference/lib/ReferenceOrderCombiner.sol [hunk 2]
```diff
@@ -481,8 +483,11 @@ contract ReferenceOrderCombiner is
...
+                    // increment total filtered executions.
+                    // totalFilteredExecutions starts at 0, infeasible to increment > 2^256.
```
Comment (0age) [inline reference/lib/ReferenceOrderCombiner.sol:488]: One tiny nit — let's ensure that comments do not cause the 80-character line limit to be exceeded

**Review by 0age — CHANGES_REQUESTED** [review]

File: reference/lib/ReferenceOrderCombiner.sol (hunk 2 above)
Comment (0xBeans) [reply reference/lib/ReferenceOrderCombiner.sol]: Fixed :)

File: reference/lib/ReferenceOrderCombiner.sol [hunk 3]
```diff
@@ -238,9 +238,11 @@ contract ReferenceOrderCombiner is
             // Otherwise, track the order hash in question.
             orderHashes[i] = orderHash;
 
-            // Decrement the number of fulfilled orders.
-            maximumFulfilled--;
-
+            unchecked {
```
Comment (0age) [inline reference/lib/ReferenceOrderCombiner.sol:241]: Tiny nit: move the comment as to why unchecked is safe to right before the unchecked block and revise to `// Skip overflow check as maximumFulfilled is nonzero.`

File: reference/lib/ReferenceOrderCombiner.sol [hunk 4]
```diff
@@ -507,8 +512,11 @@ contract ReferenceOrderCombiner is
 
             // If offerer and recipient on the execution are the same...
             if (execution.item.recipient == execution.offerer) {
-                // increment total filtered executions.
-                totalFilteredExecutions += 1;
+                unchecked {
+                    // increment total filtered executions.
+                    // executions start at 0, infeasible to increment > 2^256.
```
Comment (0age) [inline reference/lib/ReferenceOrderCombiner.sol:517]: Nit: Capitalize Executions and place that word at the end of the comment line above.

File: reference/lib/ReferenceOrderCombiner.sol [hunk 5]
```diff
@@ -764,8 +772,11 @@ contract ReferenceOrderCombiner is
...
```
Comment (0age) [inline reference/lib/ReferenceOrderCombiner.sol:777]: Nit: Capitalize Executions and place that word at the end of the comment line above.

**Review by 0age — CHANGES_REQUESTED** [review]

File: reference/lib/ReferenceOrderCombiner.sol (hunk 5 above)
Comment (0xBeans) [reply reference/lib/ReferenceOrderCombiner.sol:777]: Do you mean this?

```
                    // increment total filtered Executions.
                    // starts at 0, infeasible to increment > 2^256 Executions.
```

File: reference/lib/ReferenceOrderCombiner.sol (hunk 3 above)
Comment (0xBeans) [reply reference/lib/ReferenceOrderCombiner.sol:242]: Done.

File: reference/lib/ReferenceOrderCombiner.sol (hunk 5 above)
Comment (0xBeans) [reply reference/lib/ReferenceOrderCombiner.sol]: Changed comment to `// Executions start at 0, infeasible to increment > 2^256.` and moved above unchecked{} block and capitlized `increments` as that seems to be the standard among the entire code base.

**Comment by 0xBeans** [comment]
@0age Capitalized `increment` comment as the entire codebase seems to follow that structure

Screen Shot 2022-06-06 at 6 07 27 PM

**Comment by hrkrshnn** [comment]
I assumed adding `unchecked` into ReferenceOrdercombiner was against the design--keeping Reference without any optimizations. If we should be pedantic about it, this change should be reverted. cc @0age

**Pull request merged**